.github/workflows/cache.yml

```diff
@@ -0,0 +1,23 @@
+name: Delete Caches
+
+# yamllint disable-line rule:truthy
+on:
+  pull_request:
+    types: [closed]
+  workflow_dispatch:
+    inputs:
+      pr-number:
+        description: "Closed PR number"
+        type: string
+        required: true
+
+jobs:
+  delete:
+    name: "Delete unused caches (PR #${{ github.event.inputs.pr-number || github.event.number }})"
+    runs-on: ubuntu-latest
+    permissions:
+      actions: write
+    steps:
+      - uses: snnaplab/delete-branch-cache-action@v1.0.0
+        with:
+          ref: refs/pull/${{ github.event.inputs.pr-number || github.event.number }}/merge
```

Comment:
That external github action is a composite with a single task:

<https://github.com/snnaplab/delete-branch-cache-action/blob/main/action.yml#L27>

Should we just inline that one instead of relying on an external action?
Especially considering the logic of that action is also added to `ci.yaml` in this same PR.

Reply:
> Should we just inline that one instead of relying on an external action? Especially considering the logic of that action is also added to `ci.yaml` in this same PR.

It's similar but I couldn't reuse the action which is why I inlined and modified it initially.
Pushed a new commit which inlines the other job, too. Included some minor improvements as well.

Tested both jobs on my fork:
https://github.com/cdce8p/ha-core/actions/runs/3483406693/jobs/5826924766#step:2:1
https://github.com/cdce8p/ha-core/actions/runs/3483455996/jobs/5826964696#step:2:1